rfcs/41-ui-navigation.md

```diff
@@ -0,0 +1,394 @@
+
+Here is a draft of how I could specify in-code how to navigate this menu.
+`build_ui!` is a theoretical macro that builds a bevy UI without the verbosity
+we currently are accustomed to.
+
+Think of `horizontal`, `vertical`, `panel` etc. as a `NodeBundle`s presets, the
+elements between `{}` are additional `Component`s added to the `NodeBundle` and
+elements within parenthesis following a keyword are children `Entity` added with
+`.with_children(|xyz| {...})`.
+
+```rust
+build_ui! {
+  vertical {:container "menu"} (
+    horizontal(
+      tab {:navigable} ("soul")
+      tab {:navigable} ("body")
+      tab {:navigable} ("mind")
+      tab {:navigable} ("all")
+    )
+    horizontal {:container "soul menu"} (
+      vertical(
+        panel {:navigable} (
+          colored_circle(Color::BLUE)
+          title_label("gdk")
+        )
+        panel {:navigable} (
+          colored_circle(Color::GREEN)
+          title_label("kfc")
+        )
+        panel {:navigable} (
+          colored_circle(Color::BEIGE)
+          title_label("abc")
+        )
+      )
+      vertical {:container "abc menu"} (
+        title_label("ABC")
+        grid(
+          circle_letter {:navigable} ("a")    circle_letter {:navigable} ("A")
+          circle_letter {:navigable} ("b")    circle_letter {:navigable} ("B")
+          circle_letter {:navigable} ("c")    circle_letter {:navigable} ("C")
+        )
+      )
+    )
+  )
+}
+fn react_events(events: ResMut<Event<NavEvent>>, game: Res<Game>) {
+  for event in events.iter() {
+    if matches!(event, NavEvent::Caught(NavCommand::Action, game.ui.start_button)) {
+      // start game
+    }
+    //etc.
+  }
+}
+```
+
+
+## Implementation strategy
+
+### Basic physical navigation
+
+Let's drop the idea to change submenu with LT/RT for a while. We would change
+submenus by navigating upward with the Dpad or arrow keys to the tab bar and
+selecting the tab with the LEFT and RIGHT keys.
+
+Typical game engine implementation of menu navigation requires the developer to
+specify the "next", "left", "right", "previous" etc.[^1] relationships between
+focusable elements. This is just lame! Our game engine not only already knows
+the position of our UI elements, but it also has knowledge of the logical
+organization of elements through the `Parent`/`Children` relationship. (in this
+case, children are specified in the parenthesis following the element name,
+look at `grid(...)`)
+
+Therefore, with the little information we provided through this hand wavy
+specification, we should already have everything setup for the navigation to
+just work™.
+
+
+### Dimensionality
+
+Let's go back to our ambition of using LT/RT for navigation now. 
+
+UI navigation is not just 2D, it's N-dimensional. The
+LT/RT to change tabs is like navigating in a 3rd orthogonal dimension to the
+dimensions you navigate with UP DOWN LEFT RIGHT inside the menus.
+
+I'll call those dimensions `Plane`s because it's easier to type than
```

Comment:
Oh I see. `Plane` is not a great word for this: it implies multidimensionality, and doesn't work for looping behavior, which is both common and important.

`Axis` is probably the most common word that could be used here.